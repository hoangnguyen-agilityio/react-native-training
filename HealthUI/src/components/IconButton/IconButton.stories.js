import { storiesOf } from '@storybook/react-native';
import React from 'react';
import IconButton from '.';
import CenterView from '../CenterView';
import ArrowIcon from '../../assets/right-arrow.svg';
import AddIcon from '../../assets/add.svg';

storiesOf('IconButton', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('With next icon', () => (
    <IconButton>
      <ArrowIcon />
    </IconButton>
  ))
  .add('With add icon', () => (
    <IconButton>
      <AddIcon />
    </IconButton>
  ));
