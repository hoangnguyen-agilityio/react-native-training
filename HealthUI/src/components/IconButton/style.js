import colors from "../../constants/colors";

export default {
    button: {
      backgroundColor: colors.bg_purple,
      color: colors.white,
      padding: 16,
      borderRadius: 50,
    }
}