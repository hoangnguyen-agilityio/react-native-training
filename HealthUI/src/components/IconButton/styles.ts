import { StyleSheet } from 'react-native';
import colors from '../../constants/colors';

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.purple,
    color: colors.white,
    padding: 16,
    borderRadius: 50,
  },
});

export default styles;
