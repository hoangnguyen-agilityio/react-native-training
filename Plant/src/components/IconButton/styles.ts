import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  button: {
    padding: 17,
    borderRadius: 50,
  },
  shadow: {
    shadowColor: 'black',
    shadowOffset: {
      width: 4,
      height: 4,
    },
    shadowOpacity: 0.37,
    shadowRadius: 24,
    elevation: 20,
  },
});

export default styles;
