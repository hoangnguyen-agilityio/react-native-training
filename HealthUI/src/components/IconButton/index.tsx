import React from 'react';
import { TouchableHighlight } from 'react-native';
import style from './style';

const IconButton = (props: any) => {
  return <TouchableHighlight style={style.button} {...props} />;
};

export default IconButton;
