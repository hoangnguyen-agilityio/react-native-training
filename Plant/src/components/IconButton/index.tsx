import React, { FC, ReactNode } from 'react';
import { TouchableOpacity } from 'react-native';
import styles from './styles';
import style from './styles';

interface Props {
  color: string;
  children: ReactNode;
  onPress?: () => void;
  shadow?: boolean;
}

const IconButton: FC<Props> = ({
  color,
  children,
  onPress = () => {},
  shadow = false,
}) => {
  return (
    <TouchableOpacity
      style={[
        style.button,
        shadow ? styles.shadow : null,
        { backgroundColor: color },
      ]}
      onPress={onPress}>
      {children}
    </TouchableOpacity>
  );
};

export default IconButton;
