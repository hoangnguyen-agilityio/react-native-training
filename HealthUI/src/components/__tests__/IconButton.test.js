import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from '../IconButton';
import { Text } from 'react-native';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <IconButton>
        <Text>
          Test Icon Button
        </Text>
      </IconButton>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
