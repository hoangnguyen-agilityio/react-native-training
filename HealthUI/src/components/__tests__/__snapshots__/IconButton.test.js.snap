// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5142AB",
      "borderRadius": 50,
      "color": "#FFFFFF",
      "padding": 16,
    }
  }
>
  <Text>
    Test Icon Button
  </Text>
</View>
`;
